example_dags/ray_taskflow_example_dynamic_config.py

```diff
@@ -0,0 +1,59 @@
+from datetime import datetime
+from pathlib import Path
+
+from airflow.decorators import dag, task
+
+from ray_provider.decorators.ray import ray
+
+
+def generate_config():
+    CONN_ID = "ray_conn"
+    RAY_SPEC = Path(__file__).parent / "scripts/ray.yaml"
+    FOLDER_PATH = Path(__file__).parent / "ray_scripts"
+    return {
+        "conn_id": CONN_ID,
+        "runtime_env": {"working_dir": str(FOLDER_PATH), "pip": ["numpy"]},
+        "num_cpus": 1,
+        "num_gpus": 0,
+        "memory": 0,
+        "poll_interval": 5,
+        "ray_cluster_yaml": str(RAY_SPEC),
+        "xcom_task_key": "dashboard",
+    }
+
+
+@dag(
+    dag_id="Ray_Taskflow_Example_Dynamic_Config",
+    start_date=datetime(2023, 1, 1),
+    schedule=None,
+    catchup=False,
+    tags=["ray", "example"],
+)
+def ray_taskflow_dag():
+
+    @task
+    def generate_data():
+        return [1, 2, 3]
+
+    @ray.task(config=generate_config)
```

Comment:
Not sure this is dynamic .. for instance would we be able to reference the airflow context to build out a config that uses the dag_run_id?
```suggestion
    @ray.task(config=generate_config(context))
```

Reply:
If its only for uniqueness of the cluster name, can we use any UUID?